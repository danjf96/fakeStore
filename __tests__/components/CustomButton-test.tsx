/**
 * @format
 */

 import 'react-native';
 import React from 'react';
 import CustomButton from '../../src/components/customButton';
 
 // Note: test renderer must be required after react-native.
import { fireEvent, render } from 'react-native-testing-library';
 
describe('renders correctly', () => {
    const onPress = jest.fn()
    const component = <CustomButton text='BUTTON' onPress={onPress} testID='button'/>
    it('basic', () => {
        const screen = render(component);
        expect(screen.toJSON()).toMatchSnapshot()
    });

    it('onPress and text', () => {
        const { getByText, getByTestId } = render(component);
        expect(getByText('BUTTON')).toBeTruthy()
        fireEvent(getByTestId('button'),'onPress')
        expect(onPress).toBeCalled()
    });
})

 