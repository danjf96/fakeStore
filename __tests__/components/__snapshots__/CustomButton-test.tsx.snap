// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly basic 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#504DB6",
      "borderRadius": 37,
      "flexDirection": "column",
      "height": 50,
      "justifyContent": "center",
      "opacity": 1,
      "width": "100%",
    }
  }
  testID="button"
>
  <Text
    style={
      Array [
        Object {
          "color": "white",
          "fontFamily": "WorkSans-Regular",
          "fontSize": 16,
          "fontWeight": "700",
        },
        undefined,
      ]
    }
  >
    BUTTON
  </Text>
</View>
`;
